src/components/Files/FileFolderSelect.tsx

```diff
@@ -0,0 +1,278 @@
+      if (!folderId) {
+        setIsAllFilesOpened((value) => !value);
+        setOpenedFoldersIds([]);
+        setSelectedFolderId(folderId);
+        return;
+      }
+
+      if (openedFoldersIds.includes(folderId)) {
+        const childFolders = getChildAndCurrentFoldersIdsById(
+          folderId,
+          folders,
+        );
+        setOpenedFoldersIds(
+          openedFoldersIds.filter((id) => !childFolders.includes(id)),
+        );
+      } else {
+        setOpenedFoldersIds(openedFoldersIds.concat(folderId));
+        dispatch(FilesActions.getFolders({ path: folderId }));
+      }
+    },
+    [dispatch, folders, openedFoldersIds],
+  );
+
+  const handleFolderSelect = useCallback(
+    (folderId: string) => {
+      setSelectedFolderId(folderId);
+
+      handleToggleFolder(folderId);
+    },
+    [handleToggleFolder],
+  );
+
+  const handleAddFolder = useCallback(
+    (parentFolderId: string) => {
+      dispatch(FilesActions.addNewFolder({ relativePath: parentFolderId }));
+
+      if (!openedFoldersIds.includes(parentFolderId)) {
+        setOpenedFoldersIds(openedFoldersIds.concat(parentFolderId));
+        dispatch(FilesActions.getFolders({ path: parentFolderId }));
+      }
+    },
+    [dispatch, openedFoldersIds],
+  );
+  const handleRenameFolder = useCallback(
+    (newName: string, folderId: string) => {
+      const folderWithSameName = folders.find(
+        (folder) => folder.name === newName,
+      );
+
+      if (folderWithSameName) {
+        dispatch(
+          UIActions.showToast({
+            message: t(`Not allowed to have folders with same names`),
+            type: 'error',
+          }),
+        );
+        return;
+      }
+      dispatch(FilesActions.renameFolder({ folderId, newName }));
+    },
+    [dispatch],
+  );
+
+  return (
+    <FloatingPortal id="theme-main">
+      {isOpen && (
+        <FloatingOverlay
+          lockScroll
+          className="z-50 flex items-center justify-center bg-gray-900/70 p-3 dark:bg-gray-900/30"
+        >
+          <FloatingFocusManager context={context}>
+            <div
+              className="relative flex max-h-full min-w-full flex-col gap-4 rounded bg-gray-100 p-6 dark:bg-gray-700 md:min-w-[425px] md:max-w-full"
+              ref={refs.setFloating}
+              {...getFloatingProps()}
+            >
+              <button
+                className="absolute right-2 top-2"
+                onClick={() => onClose(false)}
+              >
+                <IconX className="text-gray-500" />
+              </button>
+              <div className="flex flex-col gap-2 overflow-auto">
+                <div className="flex justify-between">
+                  <h2 id={headingId} className="text-base font-semibold">
+                    {t('Select folder')}
+                  </h2>
+                </div>
+                {folders.length === 0 &&
+                ['LOADING', undefined].includes(foldersStatus) ? (
```

Comment:
Move `['LOADING', undefined]` in constant outside the component to not create array each time and wrap into `Set`
```
const loadingStatuses = new Set(['LOADING', undefined]);
```
and use
```suggestion
                loadingStatuses.has(foldersStatus) ? (
```